### PR TITLE
Update of CoW AMM solver tutorial

### PR DESCRIPTION
**Pull request by fhenneke**
# Description

This is a quick update of the solver tutorial on integrating Balancer CoW AMMs.

I am not really able to speak about the technical details. So it would be good if someone from backend/smart contract could have a look.

The goal of the PR is to remove wrong information. Adding a good tutorial can be done later.

# Changes

This PR mostly removes information which was outdated. In some places it adds new links, contract names, functions.

### CODE REVIEW

File: docs/cow-protocol/tutorials/solvers/cow-amm.md [hunk 1]
```diff
@@ -63,35 +38,35 @@ You need to choose a valid CoW Swap order with the following restrictions:
...
-However, leaving the commit set means that no AMM orders will appear in the orderbook until the commit is reset.
-Not clearing the commit at the end of the batch is considered slashable behavior.
+This order can be included in a batch as any other CoW Protocol orders with two extra conditions:
+- One of the pre-interaction must set the commitment by calling `BCoWPool.commit(hash)`.
+- Must contain at most one order from the AMM in the same batch.
 
 ### Surplus
 
-The surplus for a CoW AMM order is measured differently depending on which AMM order is executed.
+The surplus for a CoW AMM order is measured exactly as for other orders by comparing the traded amounts to limit amounts.
```
Comment (bram-vdberg) [inline docs/cow-protocol/tutorials/solvers/cow-amm.md]:
```suggestion
The surplus for a CoW AMM order is measured the same way as other orders, by comparing the traded amounts to limit amounts.
```

File: docs/cow-protocol/tutorials/solvers/cow-amm.md [hunk 2]
```diff
@@ -63,35 +38,35 @@ You need to choose a valid CoW Swap order with the following restrictions:
...
-- One of the pre-interaction must set the commitment by calling `ConstantProduct.commit(hash)`.
-- Must contain at most one order from the AMM in the same batch.
-- One of the post-interactions must reset the commitment by calling `ConstantProduct.commit(EMPTY_COMMITMENT)`.
+Example code that generates an order can be found in `order` function of the BCoWHelper contract
 
-The last step (clearing the commit) is technically not required for the batch to settle successfully, however it makes the settlement overall cheaper, since it resets the storage slot.
-However, leaving the commit set means that no AMM orders will appear in the orderbook until the commit is reset.
-Not clearing the commit at the end of the batch is considered slashable behavior.
+This order can be included in a batch as any other CoW Protocol orders with two extra conditions:
+- One of the pre-interaction must set the commitment by calling `BCoWPool.commit(hash)`.
```
Comment (bram-vdberg) [inline docs/cow-protocol/tutorials/solvers/cow-amm.md]:
```suggestion
- One of the pre-interaction steps must set the commitment by calling `BCoWPool.commit(hash)`.
```

**Review by bram-vdberg — APPROVED** [review]
Looks good to me!

File: docs/cow-protocol/tutorials/solvers/cow-amm.md [hunk 3]
```diff
@@ -63,35 +38,35 @@ You need to choose a valid CoW Swap order with the following restrictions:
 - `sellTokenBalance`: must be `BALANCE_ERC20`.
 - `buyTokenBalance`: must be `BALANCE_ERC20`.
 
+:::note
+
+The sell and buy amount specified are the limit amounts of the order. The actual executed amounts are determined using prices from the auction.
+In that regard, CoW AMM orders behave exactly like normal trader orders.
```
Comment (fleupold) [inline docs/cow-protocol/tutorials/solvers/cow-amm.md:44]: Can we add that if solvers are using non co-located drivers, they should encode this trade as a trade of `kind=jit` (specifying executed amount and fee amount)?

File: docs/cow-protocol/tutorials/solvers/cow-amm.md [hunk 4]
```diff
@@ -63,35 +38,35 @@ You need to choose a valid CoW Swap order with the following restrictions:
...
 You also need to compute:
 - the order hash `hash` as defined in the library `GPv2Order`, and
-- the order signature (example code that generates a valid signature is [the `getTradeableOrderWithSignature` function in the ComposableCoW repo](https://github.com/cowprotocol/composable-cow/blob/main/src/ComposableCoW.sol)).
+- the order signature
 
-This order can be included in a batch as any other CoW Protocol orders with three extra conditions:
-- One of the pre-interaction must set the commitment by calling `ConstantProduct.commit(hash)`.
-- Must contain at most one order from the AMM in the same batch.
-- One of the post-interactions must reset the commitment by calling `ConstantProduct.commit(EMPTY_COMMITMENT)`.
+Example code that generates an order can be found in `order` function of the BCoWHelper contract
```
Comment (fleupold) [inline docs/cow-protocol/tutorials/solvers/cow-amm.md]: Can we link here the factory and helper contract deployment addresses here (cf. https://cowservices.slack.com/archives/C035TDPHQAJ/p1723126963478129?thread_ts=1723034842.053739&cid=C035TDPHQAJ)

**Review by fleupold — APPROVED** [review]
Looks good.

File: docs/cow-protocol/tutorials/solvers/cow-amm.md (hunk 4 above)
Comment (fhenneke) [reply docs/cow-protocol/tutorials/solvers/cow-amm.md]: I added a link to the code of contract.

File: docs/cow-protocol/tutorials/solvers/cow-amm.md (hunk 3 above)
Comment (fhenneke) [reply docs/cow-protocol/tutorials/solvers/cow-amm.md:44]: I added a reference to the API specification for jit trades and and preinteractions at the bottom, after the discussion of `commit`.

**Comment by fleupold** [comment]
What's the process for getting this merged? I'd like to share this with solvers

**Review by cowmarketing — APPROVED** [review]

**Pull request merged**